### ISSUE TITLE
[weather-api-widget] 3-day forecast use night variants icons during the day

### ISSUE THREAD

**Issue by xfzv** [streetturtle/awesome-wm-widgets#459]
[Original conversation](https://github.com/Ryuno-Ki/awesome-wm-widgets/commit/449f2dbb3944e58ebe6a97ac87bf5137b76d255a)

Opening this issue as requested in https://github.com/Ryuno-Ki/awesome-wm-widgets/commit/449f2dbb3944e58ebe6a97ac87bf5137b76d255a#commitcomment-148523178

---

With

```lua
icons = "weather-underground-icons",
icons_extension = ".png",
show_daily_forecast = true,
```

The night variant icons are used for the 3-day forecast during day time (and probably the other way around, unconfirmed for now):

![1](https://github.com/user-attachments/assets/09cc0d31-034f-48a0-97ae-81d0ba96c446)

Reverting https://github.com/Ryuno-Ki/awesome-wm-widgets/commit/449f2dbb3944e58ebe6a97ac87bf5137b76d255a solves the issue but then the current condition icon (in the bar and in the preview) is using the wrong variant:

![2](https://github.com/user-attachments/assets/e3aa7e8d-a623-44c6-9347-fd01313ff48e)

**Comment by Ryuno-Ki** [comment]
Will look into it once #458 is merged.

**Comment by Ryuno-Ki** [comment]
I had a rare moment of spare time + clear weather forecast and went ahead to write a fix in #464.

**Comment by Ryuno-Ki** [comment]
Not easily. What you could do is guessing that the future will behave similar to today and [look at the current date](https://github.com/streetturtle/awesome-wm-widgets/blob/d6733cf2b686f60ff9fc0e59fd8dc960e2d3b3f8/weather-api-widget/example_response.json#L17).

This would [slightly increase the function signature](https://github.com/streetturtle/awesome-wm-widgets/blob/d6733cf2b686f60ff9fc0e59fd8dc960e2d3b3f8/weather-api-widget/weather.lua#L617).

**Comment by Ryuno-Ki** [comment]
Once it is I can create another PR (marked as Draft) for people to try out and provide feedback.
